site/gatsby-site/server/tests/utils.ts

```diff
@@ -38,30 +39,198 @@ export const seedCollection = async ({ name, docs, database = 'aiidprod', drop =
     }
 }
 
-interface AuthResponse {
-    access_token: string;
-    refresh_token: string;
-    user_id: string;
-}
+export const seedUsers = async (users: { userId: string, roles: string[] }[], { drop } = { drop: true }) => {
 
-export const login = async (username: string, password: string) => {
-
-    const response = await fetch(`https://services.cloud.mongodb.com/api/client/v2.0/app/${config.REALM_APP_ID}/auth/providers/local-userpass/login`, {
-        method: 'POST',
-        headers: {
-            'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({
-            username,
-            password,
-        })
+    await seedCollection({
+        name: 'users',
+        database: 'customData',
+        docs: users,
+        drop,
     });
+}
 
-    if (!response.ok) {
-        throw new Error(`Error login in! \n\n ${await response.text()}`);
-    }
+export const makeRequest = async (url: string, data: { variables?: Record<string, unknown>, query: string }) => {
 
-    const data: AuthResponse = await response.json();
+    return supertest(url)
+        .post('/')
+        .set('Authorization', `Bearer dummyToken`)
+        .send(data);
+}
 
-    return data;
+export function serializeId<T extends { _id?: { toHexString: () => string } }>(obj: T): T { return ({ ...obj, _id: obj._id?.toHexString() }) }
+
+export function removeId<T extends { _id?: { toHexString: () => string } }>(obj: T): T { return ({ ...obj, _id: undefined }) }
+
+
+/**
+ * Interface representing the structure of a fixture for testing api operations.
+ * 
+ * @template T - The database object type of the main entity being tested. This usually coincides with the graphql object type, but sometimes they may differ like with Incidents.
+ * @template Y - The generated graphql object type of the main entity being tested.
```

Comment:
Document this template parameter as optional as we discussed, for the explicit case where the DB object matches the GraphQL output object we return from the API.